### ISSUE TITLE
filtering false positive graph link/paths

### ISSUE THREAD

**Issue by XuewenWangUGA** [pangenome/pggb#27]
I found that the pggb output graph has too many false positive links to the other places in the genomes.
 Is there any better way to filter out the long link and keep appropriate links close to the gaps? Detailed instructions of how to tune the parameters should be very useful.
My thing is to keep some links among very close blocks and filter out those across many blocks. The genomes are collinearity so the links among very close blocks should be the primary ones and should be kept. My genome may have many repeat sequences. Some links (edge) are obviously wrong which are several mega bp long and across many blocks. 

used parameters: 
-s25000-p70-n5-a70-K16-k5-w10000-j5000-e5000
graph image: [https://ibb.co/m9TmGGK](url)

**Comment by ekg** [comment]
The key thing to reduce these is to use a larger -s segment length. In the
past this has led to very slow alignment times, so I have not recommended
going beyond 100kb. However, we seem to get simpler graphs by forcing even
longer segment lengths, so there is an advantage to increasing this
threshold. It has been technically difficult due to the quadratic scaling
of pairwise alignment.

To resolve this issue I have developed a new alignment algorithm for very
long sequence pairs that will let us increase -s as far as needed to remove
spurious links in a given graph, probably to the point of requiring full
length alignment between whole chromosomes.

Next week the pipeline will be updated to support this. The command line
may change slightly. Thanks for the feedback!

**Comment by XuewenWangUGA** [comment]
Eric,
Thanks for your response. Look forward to your new update and I am happy to test it.
In addition, you can update this inconsistency:
In the github instruction it states "for only travers a given block for -W[N], --min-subpath=[N] bp", but the tool use -P for --min-subpath=[N] bp in smoothxg. Which one is the correct parameter?

**Comment by XuewenWangUGA** [comment]
Minimap2 mapping/alignment tool can do very long sequence alignment at the chromosome level

**Comment by ekg** [comment]
minimap2 is an alternative (pggb was initially based on it), but it is slow
for long sequence pairs, uses a lot of memory, and it is not possible to
enforce the global alignment of the sequences. It had also had trouble in
the chaining step in regions with high repetition, although I understand
this may have been resolved.

It is seeded alignment based on heuristic chaining of minimizers. This
means that if you use minimap2 you will need to filter short alignments or
otherwise recompute the pieces of the optimal alignment. In my experience,
this will tend to break apart the pangenome graph at thresholds that are
required to remove spurious alignments, making this alignment length
filtering rather tricky.

I am sure that it could be adapted to some extent, but the all to all
alignment is necessarily expensive and so I have worked to optimize this
step. Also, the requirements of the alignment are different. Because we
realign the graph locally (in smoothxg at the end of the pipeline) we can
use lower-cost edit distance based alignment as a kind of scaffolding step.
In the end we will use a more costly convex gap model to derive the base
level alignment of the sequences in each region of the graph.
